doc/guides/writing-tests.md

```diff
@@ -312,7 +312,7 @@ TEST_F(EnvTest, RunAtExit) {
   v8::Local<v8::Context> context = v8::Context::New(isolate_);
   node::IsolateData* isolateData = node::CreateIsolateData(isolate_, uv_default_loop());
   Argv argv{"node", "-e", ";"};
-  auto env = Environment:CreateEnvironment(isolateData, context, 1, *argv, 2, *argv);
+  auto env = node::CreateEnvironment(isolateData, context, 1, *argv, 2, *argv);
   node::AtExit(at_exit_callback);
```

Comment:
LGTM, could you also pass the `env` to the AtExit function:
```c++
node::AtExit(env, at_exit_callback);
```
Without this AtExit will use the [thread local environment](https://github.com/nodejs/node/blob/c2d738db8053a0e090aabbeeb088b34c3da7b7ad/src/node.cc#L4322) which will be different and cause a segment fault. I'll take a closer look into the exact cause of this, or let me know if you'd like to dig into it further.